File: Leetcode(Swift)/String_related/Leetcode_3.swift
/*
 Given a string s, find the length of the longest substring without repeating characters.
 */

import Foundation

final class Solution_3 {
    
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var maxCount = 0
        var array = [Character]()
        for char in s {
            if let index = array.firstIndex(of: char) {
                array.removeFirst(index + 1)
            }
            array.append(char)
            if(array.count > maxCount){
                maxCount = array.count
            }
        }
        return maxCount
    }
}

File: 算法分类/数组-字符串-链表/滑动窗口/leetcode_3.py

# Given a string s, find the length of the longest substring without repeating characters.

class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        hashmap = {}
        left = maxLen = 0
        for i, c in enumerate(s):
            if c in hashmap:
                left = max(left, hashmap[c] + 1)
                print(left)
            hashmap[c] = i
            maxLen = max(maxLen, i - left + 1)
        return maxLen


s = "abba"
so = Solution()

r = so.lengthOfLongestSubstring(s)

print(r)